processor/k8sprocessor/processor.go

```diff
@@ -92,56 +93,31 @@ func (kp *kubernetesprocessor) ProcessMetrics(ctx context.Context, md pdata.Metr
 func (kp *kubernetesprocessor) ProcessLogs(ctx context.Context, ld pdata.Logs) (pdata.Logs, error) {
 	rl := ld.ResourceLogs()
 	for i := 0; i < rl.Len(); i++ {
-		kp.processResource(ctx, rl.At(i).Resource(), k8sIPFromAttributes())
+		kp.processResource(ctx, rl.At(i).Resource())
 	}
 
 	return ld, nil
 }
 
-func (kp *kubernetesprocessor) processResource(ctx context.Context, resource pdata.Resource, attributeExtractors ...ipExtractor) {
-	var podIP string
+func (kp *kubernetesprocessor) processResource(ctx context.Context, resource pdata.Resource) {
 
-	for _, extractor := range attributeExtractors {
-		podIP = extractor(resource.Attributes())
-		if podIP != "" {
-			break
-		}
-	}
-
-	// Check if the receiver detected client IP.
-	if podIP == "" {
-		if c, ok := client.FromContext(ctx); ok {
-			podIP = c.IP
-		}
-	}
-
-	// If IP is still not available by this point, nothing can be tagged here. Return.
-	if podIP == "" {
+	podIdentifiers := k8sPodAssociationFromAttributes(ctx, resource.Attributes(), kp.podAssociations)
+	if len(podIdentifiers) == 0 {
 		return
 	}
-
-	resource.Attributes().InsertString(k8sIPLabelName, podIP)
-
-	// Don't invoke any k8s client functionality in passthrough mode.
-	// Just tag the IP and forward the batch.
-	if kp.passthroughMode {
-		return
-	}
-
-	// add k8s tags to resource
-	attrsToAdd := kp.getAttributesForPodIP(podIP)
-	if len(attrsToAdd) == 0 {
-		return
-	}
-
-	attrs := resource.Attributes()
-	for k, v := range attrsToAdd {
-		attrs.InsertString(k, v)
+	for k, v := range podIdentifiers {
+		if !kp.passthroughMode {
+			attrsToAdd := kp.getAttributesForPod(v)
+			for key, val := range attrsToAdd {
+				resource.Attributes().InsertString(key, val)
+			}
+		}
+		resource.Attributes().InsertString(k, v)
```

Comment:
Previously in passthroughMode we were only adding pod ip. Now it seems we are adding everything we know about the pod. Why the change in logic? Is this intentional?

Reply:
@pmatyjasek-sumo Can you clarify this please?

Reply:
It behaves the same as previous. Pod attributes are added only when `passthtoughMode` is off. Line 114 adds only identifier to batch. Same as previously.